core/src/replay_stage.rs

```diff
@@ -3074,6 +3074,37 @@ impl ReplayStage {
 
                                 tower.vote_state.root_slot = bank_vote_state.root_slot;
                                 tower.vote_state.votes = bank_vote_state.votes;
+
+                                let last_voted_slot = tower.vote_state.last_voted_slot().unwrap_or(
+                                    // If our local root is higher than the highest slot in `bank_vote_state` due to
+                                    // supermajority roots, then it's expected that the vote state will be empty.
```

Comment:
not exactly clear to me how supermajority roots cause tower to be empty

Reply:
you're right I don't think wait for super majority triggers this. it would be cases where root is artificially increased, which i believe could only happen when restarting from a snapshot. I'll update the comment.

Reply:
Anything that artificially increases the root could trigger this. I think snapshot is the best example, but I think supermajority roots could also cause this:
https://github.com/solana-labs/solana/blob/d90e1582869d8ef8d386a1c156eda987404c43be/ledger/src/blockstore_processor.rs#L1576-L1580